File: 03-poo/05-relaciones-entre-objetos/scala/soluciones/RetoPedido.scala
case class Cliente(nombre: String)
case class DetallePedido(producto: String, cantidad: Int)
class Pedido(val cliente: Cliente):
  private var detalles = List.empty[DetallePedido]
  def agregar(producto: String, cantidad: Int): Unit = detalles ::= DetallePedido(producto, cantidad)
  def totalDetalles: Int = detalles.size

@main def retoPedido(): Unit =
  val p = new Pedido(Cliente("Cliente Chilete"))
  p.agregar("Cuaderno", 2)
  println(p.totalDetalles)

File: 03-poo/05-relaciones-entre-objetos/java/soluciones/RetoPedido.java
import java.util.ArrayList;
import java.util.List;

public class RetoPedido {
    static class Cliente { final String nombre; Cliente(String nombre) { this.nombre = nombre; } }
    static class DetallePedido { final String producto; final int cantidad; DetallePedido(String p, int c){ producto=p; cantidad=c; } }
    static class Pedido {
        // El pedido se asocia con un cliente y compone sus detalles.
        final Cliente cliente;
        final List<DetallePedido> detalles = new ArrayList<>();
        Pedido(Cliente cliente) { this.cliente = cliente; }
        void agregar(String producto, int cantidad) { detalles.add(new DetallePedido(producto, cantidad)); }
    }
    public static void main(String[] args) {
        Pedido pedido = new Pedido(new Cliente("Cliente Chilete"));
        pedido.agregar("Cuaderno", 2);
        System.out.println(pedido.detalles.size());
    }
}
